patches/tModLoader/Terraria/ModLoader/Default/DefaultBossBarStyle.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace Terraria.ModLoader.Default
+{
+	/// <summary>
+	/// This is the default boss bar style - the way vanilla draws boss bars.
+	/// </summary>
+	[Autoload(false)]
+	internal class DefaultBossBarStyle : ModBossBarStyle
```

Comment:
This class feels unnecessary at a glance - what is its purpose?
If it needs to exist temporarily, why can't it use a constructor to set name, and be discarded when done?
Alternatively, if it is just showing how to get a quick vanilla style bar, wouldn't that fit EM better?

Reply:
This is internal to tml, and is required to be in line with how ModBossBarStyle is managed in the menus. (And it's a mirror of the ModMenu design). An example for a ModBossBarStyle exists in EM.

Reply:
Thanks for clarifying, more of just questioning how its used - 
Looking at what you pointed at, it's used as a 'backing' correct? 

Then to reiterate my question of constructors, keeping in mind this comment chain ( https://github.com/tModLoader/tModLoader/pull/1288#discussion_r561351794) why are we not using a constructor such as set `string mod ="VanillaSpoof"` in the default style so FullName and TryFind() work? 
Or, for that matter, actually directly change its full name so it works without using a constructor if its always the same "fake" mod?